doc/source/testing_pull_requests.rst

```diff
@@ -0,0 +1,226 @@
+apk. We expect that the contributors has enough criteria/knowledge to perform
+the other steps mentioned, so let's begin...
+
+To create an apk from a python-for-android pull request we contemplate three
+possible scenarios:
+
+  - using python-for-android commands directly from the pull request files
+    that we want to test, without installing it (the recommended way for most
+    of the test cases)
+  - installing python-for-android using the github's branch of the pull request
+  - using buildozer and a custom app
+
+We will explain the first two methods using one of the distributed
+python-for-android test apps and we assume that you already have the
+python-for-android dependencies installed. For the `buildozer` method we also
+expect that you already have a a properly working app to test and a working
+installation/configuration of `buildozer`. There is one step that it's shared
+with all the testing methods that we propose in here...we named it
+`Common steps`.
+
+
+Common steps
+^^^^^^^^^^^^
+The first step to do it's to get a copy of the pull request, we can do it of
+several ways, and that it will depend of the circumstances but all the methods
+presented here will do the job, so...
+
+Fetch the pull request by number
+--------------------------------
+For the example, we will use `1901` for the example) and the pull request
+branch that we will use is `feature-fix-numpy`, then you will use a variation
+of the following git command:
+`git fetch origin pull/<#>/head:<local_branch_name>`, eg.::
+
+    .. codeblock:: bash
+
+        git fetch upstream pull/1901/head:feature-fix-numpy
+
+.. note:: Notice that we fetch from `upstream`, since that is the original
+          project, where the pull request is supposed to be
+
+.. tip:: The amount of work of some users maybe worth it to add his remote
+       to your fork's git configuration, to do so with the imaginary
+       github user `Obi-Wan Kenobi` which nickname is `obiwankenobi`, you
+       will do::
+
+          .. codeblock:: bash
+
+              git remote add obiwankenobi https://github.com/obiwankenobi/python-for-android.git
+
+       And to fetch the pull request branch that we put as example, you
+       would do::
+
+          .. codeblock:: bash
+
+              git fetch obiwankenobi
+              git checkout obiwankenobi/feature-fix-numpy
+
+
+Clone the pull request branch from the user's fork
+--------------------------------------------------
+Sometimes you may prefer to use directly the fork of the user, so you will get
+the nickname of the user who created the pull request, let's take the same
+imaginary user than before `obiwankenobi`::
+
+    .. codeblock:: bash
+
+        git clone -b feature-fix-numpy \
+            --single-branch \
+            https://github.com/obiwankenobi/python-for-android.git \
+            p4a-feature-fix-numpy
+
+For what it's worth it, about the command above explained line by line:
```

Comment:
Minor: something looks off in this sentence. Maybe we could try something like:
`Here's the above command explained line by line:`

Reply:
**Thanks**, this is already fixed :wink: